R/ntfy.R

```diff
@@ -18,39 +18,76 @@ ntfy_server <- function(var = "NTFY_SERVER") {
   Sys.getenv(var)
 }
 
-#' Create HTTP headers for basic authentication
-#' 
-#' @param var_username environment variable in which the ntfy username is stored
-#' @param var_password environment variable in which the ntfy password is stored
-#' 
-#' @return an object with class `request` that is passed to [httr::POST()]
+#' Get the ntfy username
+#'
+#' @param var environment variable in which the username is stored
+#'
+#' @return the username with access to the protected ntfy topic
 #' @export
-#' 
-#' @examples
-#' \dontrun{
-#' Sys.setenv("NTFY_USERNAME" = "example")
-#' Sys.setenv("NTFY_PASSWORD" = "super-secret-password")
-#' 
-#' # This generates the correct header:
-#' ntfy_authorization()
-#' #> <request>
-#' #> Headers:
-#' #>   * Authorization: Basic ZXhhbXBsZTpzdXBlci1zZWNyZXQtcGFzc3dvcmQ
-#' }
-#' 
-ntfy_authorization <- function(var_username = "NTFY_USERNAME", var_password = "NTFY_PASSWORD") {
-  username = Sys.getenv(var_username)
-  password = Sys.getenv(var_password)
-  
-  if (username == "") {
-    httr::add_headers()
+ntfy_username <- function(var = "NTFY_USERNAME") {
```

Comment:
Since this file is getting long, please move these env-related functions to a new env.R - I wonder (#13) if they should have complementary `ntfy_set_x()` functions?

Reply:
Done! And it's a good place to put all the default-setting functions too